apps/mobile/src/common/transactions/stacks-transactions.hooks.ts

```diff
@@ -0,0 +1,51 @@
+import { useCallback } from 'react';
+import { FieldValues } from 'react-hook-form';
+
+import { AccountId } from '@/models/domain.model';
+import { useAccountByIndex } from '@/store/accounts/accounts.read';
+import { useStacksSigners } from '@/store/keychains/stacks/stacks-keychains.read';
+import { useNetworkPreferenceStacksNetwork } from '@/store/settings/settings.read';
+import { bytesToHex } from '@noble/hashes/utils';
+
+import { useNextNonce } from '@leather.io/query';
+import {
+  GenerateUnsignedTransactionOptions,
+  TransactionTypes,
+  generateUnsignedTransaction,
+} from '@leather.io/stacks';
+import { stxToMicroStx } from '@leather.io/utils';
+
+export function useGenerateStxTokenTransferUnsignedTx({ fingerprint, accountIndex }: AccountId) {
+  const network = useNetworkPreferenceStacksNetwork();
+  const account = useAccountByIndex(fingerprint, accountIndex);
+  const stxSigner = useStacksSigners().fromAccountIndex(fingerprint, accountIndex)[0];
+  const stxAddress = stxSigner?.address ?? '';
+  const { data: nextNonce } = useNextNonce(stxAddress);
```

Comment:
I think it might be interesting to decouple this logic. This hook is essentially responsible for 1) preparing account specific tx details, and 2) feeding them through to the `generateUnsignedTransaction` fn.

What if I want to modify these values in another tx signing use case that isn't the form specifically? I can't compose this logic well, because it assumes I always need to do both at same time.

If these were separate, I could do something like:

```ts
// send form use case
function useGenerateStxTokenTransferUnsignedTx(account: AccountId) {
  const stxAccountSignerDetails = useStxTokenTransferDetails();
  // spread details to fn
  return (formValues) => generateUnsignedTransaction({ 
    ...stxAccountSignerDetails,
    ...parsedFormValues
  })
}
```